### PR TITLE
chore: add an internal uri

### PR DESCRIPTION
**Pull request by nicolasburtey**

### CODE REVIEW

File: lib/config.ts [hunk 1]
```diff
@@ -14,7 +19,22 @@ if (!GRAPHQL_HOSTNAME) {
   }
 }
 
+if (!GRAPHQL_HOSTNAME_INTERNAL) {
```
Comment (bodymindarts) [inline lib/config.ts:22]: Why this logic? If its not set it should exit with error or?
This is only relevant for server side right?

Comment (nicolasburtey) [reply lib/config.ts:22]: for the same reason the code block above is here, we have to have multiple dns set up for bbw for the time being.

File: lib/config.ts [hunk 2]
```diff
@@ -1,5 +1,10 @@
 let GRAPHQL_HOSTNAME = process.env.NEXT_PUBLIC_GRAPHQL_HOSTNAME as string
 
+// we need an internal dns to properly propagate the ip related headers to api
+// if we use the api endpoints, nginx will rewrite the header to prevent spoofing
+let GRAPHQL_HOSTNAME_INTERNAL = process.env.NEXT_INTERNAL_GRAPHQL_HOSTNAME as string
```
Comment (krtk6160) [inline lib/config.ts:5]:
```suggestion
let GRAPHQL_HOSTNAME_INTERNAL = process.env.NEXT_PUBLIC_GRAPHQL_HOSTNAME_INTERNAL as string
```

File: lib/config.ts [hunk 3]
```diff
@@ -14,7 +19,22 @@ if (!GRAPHQL_HOSTNAME) {
   }
 }
 
+if (!GRAPHQL_HOSTNAME_INTERNAL) {
+  if (typeof window !== "undefined") {
+    let hostParts = window.location.host.split(".")
+    if (hostParts.length <= 3) {
+      // throw new Error("Missing env variables")
+      hostParts = "pay.mainnet.galoy.io".split(".")
```
Comment (krtk6160) [inline lib/config.ts:27]: Shouldn't this also be changed?

**Review by krtk6160 — CHANGES_REQUESTED** [review]

File: lib/config.ts (hunk 3 above)
Comment (nicolasburtey) [reply lib/config.ts:27]: to what?

File: lib/config.ts (hunk 2 above)
Comment (nicolasburtey) [reply lib/config.ts:5]: this is the internal kubernetes dns that we should have. so using the PUBLIC string here doesn't really make sense

File: lib/config.ts (hunk 1 above)
Comment (nicolasburtey) [reply lib/config.ts:22]: my bad. actually yes this is only server side so this is irrelevant

File: lib/config.ts (hunk 3 above)
Comment (krtk6160) [reply lib/config.ts]: Internal dns. Otherwise rate limiting will be applied to `ln.bitcoinbeach.com`.

File: lib/config.ts (hunk 2 above)
Comment (krtk6160) [reply lib/config.ts]: The `NEXT_PUBLIC` is a [special prefix](https://nextjs.org/docs/basic-features/environment-variables#exposing-environment-variables-to-the-browser) that needs to added to the names of env vars being exposed to the browser.

**Pull request merged**